### ISSUE TITLE
Using wrong exchange name doesn't result in any error/warning message when publishing message

### ISSUE THREAD

**Issue by manuel-koch** [jandelgado/rabtap#55]
Using rabtap v1.27 to publish a message with the command like below.
The raptap output doesn't indicate whether publishing was actually successful, i.e. it states that message got published
while RabbitMQ states an error because the used exchange is unknown.

```
$ RABTAP_AMQPURI=amqp://guest:guest@localhost:5672/
$ RABTAP_APIURI=http://guest:guest@localhost:15672/api
$ rabtap --verbose pub --exchange non_existing_exchange --routingkey foo --format json rabtap-1628256743950000000.json
INFO[0000] waiting for new session ...
INFO[0000] sleeping for 0s
DEBUG[0000] publishing message to exchange 'non_existing_exchange' with routing key 'foo'
INFO[0000] got new amqp session ...
INFO[0000] publishing channel closed.
INFO[0000] session: shutting down factory (cancel)
```

While RabbitMQ server indicates an error on its console output regarding an unknown exchange:

```
rabbitmq              | 2021-08-06 13:39:44.940 [info] <0.2930.0> accepting AMQP connection <0.2930.0> (172.21.0.1:62628 -> 172.21.0.3:5672)
rabbitmq              | 2021-08-06 13:39:44.947 [info] <0.2930.0> connection <0.2930.0> (172.21.0.1:62628 -> 172.21.0.3:5672): user 'guest' authenticated and granted access to vhost '/'
rabbitmq              | 2021-08-06 13:39:44.951 [error] <0.2939.0> Channel error on connection <0.2930.0> (172.21.0.1:62628 -> 172.21.0.3:5672, vhost: '/', user: 'guest'), channel 1:
rabbitmq              | operation basic.publish caused a channel exception not_found: no exchange 'non_existing_exchange' in vhost '/'
rabbitmq              | 2021-08-06 13:39:44.953 [warning] <0.2930.0> closing AMQP connection <0.2930.0> (172.21.0.1:62628 -> 172.21.0.3:5672, vhost: '/', user: 'guest'):
rabbitmq              | client unexpectedly closed TCP connection
```

**Comment by jandelgado** [comment]
Yes, that would be a useful feature.I'll put it in the backlog. Thanks

**Comment by jandelgado** [comment]
@manuel-koch: Current version on `notify_when_publishing_to_nonexistent_exchange` branch will now have exit code "1" when publishing with errors, please retest.

**Comment by manuel-koch** [comment]
On branch `notify_when_publishing_to_nonexistent_exchange ` I verified that exitcode is non-zero when trying to publish (at least one message) to a non-existing exchange. I created multiple JSON message files, one with a valid exchange, one with a non-existing exchange:

```
$ cat *.json | RABTAP_AMQPURI=amqp://guest:guest@localhost:5672/ ./bin/rabtap-darwin-amd64 --verbose pub --format=json
DEBUG[0000] waiting for new session on amqp://guest:xxxxx@localhost:5672/
DEBUG[0000] publishing message to exchange 'test' with routing key 'foo'
DEBUG[0000] got new amqp session ...
INFO[0000] publish delay: sleeping for 0s
DEBUG[0000] publishing message to exchange 'non-existent' with routing key 'foo'
DEBUG[0000] publishing channel closed.
DEBUG[0000] waiting for confirms & returns...
ERROR[0000] publishing error: channel error: Exception (404) Reason: "NOT_FOUND - no exchange 'non-existent' in vhost '/'"
INFO[0001] Publisher ending
ERROR[0001] publish: published with errors

$ echo $?
1
```

I'm fine with the changes !

**Comment by jandelgado** [comment]
excellent, will be merged tonight ... Also have a look at the new `--confirms` and `--mandatory` options, which will help to detect more publishing problems

**Comment by jandelgado** [comment]
implemented with #62
